pkg/gameserverallocations/allocation_cache.go

```diff
@@ -262,6 +262,49 @@ func (c *AllocationCache) ListSortedGameServers(gsa *allocationv1.GameServerAllo
 	return list
 }
 
+// ListSortedGameServersPriorities sorts and returns a list of game servers based on the
+// list of Priorities.
+func (c *AllocationCache) ListSortedGameServersPriorities(gsa *allocationv1.GameServerAllocation) []*agonesv1.GameServer {
+	list := c.getGameServers()
+	if list == nil {
+		return []*agonesv1.GameServer{}
+	}
+
+	sort.Slice(list, func(i, j int) bool {
+		gs1 := list[i]
+		gs2 := list[j]
+
+		// TODO: Should we add FeatureStateAllocationFilter here too?
```

Comment:
I'm wondering if I should add in the below logic like we have in packed ListSortedGameServers, or if I should leave it out since this is case distributed ListSortedGameServersPriorities?

```
// Search Allocated GameServers first.
if gs1.Status.State != gs2.Status.State {
  return gs1.Status.State == agonesv1.GameServerStateAllocated
}
```

Reply:
I'm leaning towards not adding it in, but wanted to check in on your thoughts.

Reply:
In reality it probably doesn't really matter - it's a micro optimisation anyway, since the filter that is just going to look at allocated game servers are going to filter out everything else regardless.

So yeah - I agree leave it out.